mala/common/physical_data.py

```diff
@@ -389,6 +390,16 @@ def write_to_openpmd_iteration(self, iteration, array,
 
         dataset = io.Dataset(array.dtype, self.grid_dimensions)
 
+        if feature_to is None:
+            feature_to = array.shape[3]
+
+        if feature_to - feature_from != array.shape[3]:
```

Comment:
The `feature_from` parameter is technically redundant. We can leave it though, in order to verify that the array has the right shape.